### PR TITLE
EnumInsteadOfBoolean path to error should point to name of property, instead of top of node

### PR DESCRIPTION
**Pull request by veronicagg**
Addressing https://github.com/Azure/autorest/issues/2322

### CODE REVIEW

File: src/modeler/AutoRest.Swagger/Validation/EnumInsteadOfBoolean.cs [hunk 1]
```diff
@@ -52,7 +52,7 @@ public override IEnumerable<ValidationMessage> GetValidationMessages(SwaggerObje
         {
             if (entity.Type?.Equals(DataType.Boolean) == true)
             {
-                yield return new ValidationMessage(new FileObjectPath(context.File, context.Path), this, context.Parent.Key);
+                yield return new ValidationMessage(new FileObjectPath(context.File, context.Path.AppendProperty("name")), this, ((SwaggerParameter)entity).Name);
```
Comment (dsgouda) [inline src/modeler/AutoRest.Swagger/Validation/EnumInsteadOfBoolean.cs]: So what happens if I have something like this
```
"mybool":{
    "type":"boolean"
}
```

File: src/modeler/AutoRest.Swagger/Validation/EnumInsteadOfBoolean.cs [hunk 2]
```diff
@@ -50,23 +50,26 @@ public class EnumInsteadOfBoolean : TypedRule<SwaggerObject>
...
+                    foreach (KeyValuePair<string, Schema> property in ((Schema)entity).Properties)
                     {
-                        yield return new ValidationMessage(new FileObjectPath(context.File, context.Path.AppendProperty("properties").AppendProperty(property.Key)), this, property.Key);
+                        if (property.Value?.Type?.Equals(DataType.Boolean) == true)
+                        {
+                            yield return new ValidationMessage(new FileObjectPath(context.File, context.Path.AppendProperty("properties").AppendProperty(property.Key)), this, property.Key);
+                        }
                     }
+                } 
+                else if (entity.Type?.Equals(DataType.Boolean) == true)
```
Comment (dsgouda) [inline src/modeler/AutoRest.Swagger/Validation/EnumInsteadOfBoolean.cs]: Perfect!

**Comment by olydis** [comment]
any update on this PR

**Comment by veronicagg** [comment]
@olydis we were discussing turning off this rule, so I waited on this PR, the decision hasn't been made yet, so I'll take a look at CI and try to fix it.

**Pull request merged**